apis/python/src/tiledbsoma/collection.py

```diff
@@ -353,7 +357,8 @@ def _tiledb_open(self, mode: str = "r") -> tiledb.Group:
         as ``with self._tiledb_open() as G:``
         as well as ``G = self._tiledb_open(); ...; G.close()``.
         """
-        assert mode in ("r", "w")
+        if mode not in ["r", "w"]:
+            raise ValueError(f'expected mode to be one of "r" or "w"; got "{mode}"')
```

Comment:
you might use `!r` to automatically `repr()` the value in question, so you need not bring your own quote marks:
```suggestion
            raise ValueError(f'expected mode to be one of "r" or "w"; got {mode!r}')
```